README.md

```diff
@@ -980,37 +983,69 @@ CredentialAccessBoundary credentialAccessBoundary =
         CredentialAccessBoundary.newBuilder().addRule(rule).build();
 ```
 
+#### Common Usage Pattern
+
 The common pattern of usage is to have a token broker with elevated access generate these downscoped
 credentials from higher access source credentials and pass the downscoped short-lived access tokens
 to a token consumer via some secure authenticated channel for limited access to Google Cloud Storage
 resources.
 
-Using the CredentialAccessBoundary created above in the Token Broker:
+#### Generating Downscoped Tokens
+
+There are two ways to generate downscoped tokens using a
+CredentialAccessBoundary:
+
+* Server-side: Uses the `DownscopedCredentials` class. Each time a
+  downscoped token is needed, the client makes a call to the Security Token Service (STS).
+  This is suitable for applications that require downscoped tokens infrequently, or applications that reuse a single downscoped credential many times.
+* Client-side: Uses the `ClientSideCredentialAccessBoundaryFactory` class. This
+  approach minimizes calls to STS. The client retrieves necessary cryptographic
+  material once and then generates multiple downscoped tokens locally. This is
+  more efficient for applications that need to generate many unique downscoped tokens.
```

Comment:
Perhaps here or a different section, could highlight when a user may want to use one or the other? 

Any pros/cons between the two? Is one preferred? What general scenarios or guidelines would call for one vs the other?

Reply:
Updated. There are two factors that are mentioned: one is if the user needs many unique tokens or they can re-use a token, and the other is the CAB rule changes frequently. Depending on their use case, the users can choose one or the other.